source/errors.ts

```diff
@@ -103,32 +99,38 @@ exports.HTTPError = class extends GotError {
 		this.headers = response.headers;
 		this.body = response.body;
 	}
-};
+}
+
+export class MaxRedirectsError extends GotError {
+	redirectUrls?: string;
 
-exports.MaxRedirectsError = class extends GotError {
 	constructor(statusCode: number, redirectUrls: string, options: Options) {
 		super('Redirected 10 times. Aborting.', {}, options);
 		this.name = 'MaxRedirectsError';
 		this.statusCode = statusCode;
 		this.statusMessage = http.STATUS_CODES[this.statusCode];
 		this.redirectUrls = redirectUrls;
 	}
-};
+}
 
-exports.UnsupportedProtocolError = class extends GotError {
+export class UnsupportedProtocolError extends GotError {
 	constructor(options: Options) {
 		super(`Unsupported protocol "${options.protocol}"`, {}, options);
 		this.name = 'UnsupportedProtocolError';
 	}
-};
+}
+
+export class TimeoutError extends GotError {
+	timings: number;
+
+	event: string;
 
-exports.TimeoutError = class extends GotError {
 	constructor(error: any, timings: number, options: Options) {
```

Comment:
`timings` argument should be an object according to [http-timer](https://github.com/szmarczak/http-timer).

We could add a `Timings` interface in this project temporarily.

Reply:
Thanks a log @fnky. Could you explain how to add `Timings` interface here?
if you mean creates one, how should I build the `phases` type?

```typescript
interface Timings {
	start: string;
	socket: string;
	lookup: string;
	connect: string;
	upload: string;
	response: string;
	end: string;
	error: string;
	phases: Phases;
}

interface Phases { 
	wait: ????;
	dns: ?????;
        .......
}
```

Reply:
Actually, it should be:

```ts
interface Timings {
	start: number;
	socket: number | null;
	lookup: number | null;
	connect: number | null;
	upload: number | null;
	response: number | null;
	end: number | null;
	error: number | null;
	phases: { 
		wait: number | null;
		dns: number | null;
		tcp: number | null;
		request: number | null;
		firstByte: number | null;
		download: number | null;
		total: number | null;
	};
}
```

Since it may throw an error, the rest can be `null` except `start`.